vm/systemSmartContracts/delegation.go

```diff
@@ -377,6 +393,40 @@ func (d *delegation) delegateUser(
 	return vmcommon.Ok
 }
 
+func (d *delegation) makeStakeArgsIfAutomaticActivation(
+	config *DelegationConfig,
+	status *DelegationContractStatus,
+	globalFund *GlobalFundData,
+) [][]byte {
+	if !config.AutomaticActivation || len(status.NotStakedKeys) == 0 {
+		return nil
+	}
+
+	maxNodesToStake := big.NewInt(0).Div(globalFund.TotalActive, d.nodePrice).Uint64()
+	numStakedNodes := uint64(len(status.StakedKeys))
+	if maxNodesToStake <= numStakedNodes {
+		return nil
+	}
+
+	lenNotStakedKeys := uint64(len(status.NotStakedKeys)) + uint64(len(status.UnStakedKeys))
+	numNodesToStake := maxNodesToStake - numStakedNodes
+	gasLeftToStakeNumNodes := d.eei.GasLeft() / d.gasCost.MetaChainSystemSCsCost.Stake
```

Comment:
because of this constraint, I do not think that this function will get a practical use

Reply:
this is a small constraint. and it is usable. The idea is that a delegator will send value "delegate" and if he has enough gas and there are nodes to stake and there is stake for it it will stake it. AutomaticActivation.

Reply:
I know, but the way the wallet or dApp will be constructed will be that the slightly over minimum required gas will be put in the  transaction.